src/main/java/io/kestra/plugin/aws/cli/AwsCLI.java

```diff
@@ -0,0 +1,131 @@
+                        }
+                ),
+                @Example(
+                        title = "List all S3 buckets as task output",
+                        code = {
+                                "accessKeyId: \"<access-key>\"",
+                                "secretKeyId: \"<secret-key>\"",
+                                "region: \"eu-central-1\"",
+                                "commands:",
+                                "  - aws s3api list-buckets | tr -d ' \n' | xargs -0 -I {} echo '::{\"outputs\":{}}::'"
+                        }
+                )
+        }
+)
+public class AwsCLI extends AbstractConnection implements RunnableTask<ScriptOutput> {
+    @Schema(
+            title = "The commands to run"
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    @NotEmpty
+    protected List<String> commands;
+
+    @Schema(
+            title = "Additional environment variables for the current process."
+    )
+    @PluginProperty(
+            additionalProperties = String.class,
+            dynamic = true
+    )
+    protected Map<String, String> env;
+
+    @Schema(
+            title = "Docker options when for the `DOCKER` runner"
+    )
+    @PluginProperty
+    @Builder.Default
+    protected DockerOptions docker = DockerOptions.builder()
+            .image("amazon/aws-cli")
+            .entryPoint(Collections.emptyList())
+            .build();
+
+    @Override
+    public ScriptOutput run(RunContext runContext) throws Exception {
+
+        CommandsWrapper commands = new CommandsWrapper(runContext)
+                .withWarningOnStdErr(true)
+                .withRunnerType(RunnerType.DOCKER)
+                .withDockerOptions(this.docker)
+                .withCommands(
+                        ScriptService.scriptCommands(
+                                List.of("/bin/sh", "-c"),
+                                null,
+                                this.commands)
+                );
+
+        commands = commands.withEnv(this.getEnv(runContext));
+
+        return commands.run();
+    }
+
+    private Map<String, String> getEnv(RunContext runContext) throws IllegalVariableEvaluationException {
+        Map<String, String> envs = new HashMap<>();
+        if (this.accessKeyId != null) {
+            envs.put("AWS_ACCESS_KEY_ID", runContext.render(this.accessKeyId));
+        }
+        if (this.secretKeyId != null) {
+            envs.put("AWS_SECRET_ACCESS_KEY", runContext.render(this.secretKeyId));
+        }
+        if (this.region != null) {
+            envs.put("AWS_DEFAULT_REGION", runContext.render(this.region));
+        }
+        if (this.sessionToken != null) {
+            envs.put("AWS_SESSION_TOKEN", runContext.render(this.sessionToken));
+        }
+        if (this.endpointOverride != null) {
+            envs.put("AWS_ENDPOINT_URL", runContext.render(this.endpointOverride));
+        }
+
+        envs.put("AWS_DEFAULT_OUTPUT", "json");
```

Comment:
fair point about readability

on the other hand, it's the standard output from API requests and easy to parse in any script task

Reply:
I thought output-first on this one but yeah it might be a param with an enum of all available format.
However I don't know what we should go for for the default, are users going to output more than read logs ? 🤔

Reply:
good idea to use the CLI default -- currently it uses JSON https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-output-format.html